### DISCUSSION TITLE
360x180 degree equirectangular stitching

### DISCUSSION THREAD

**Discussion by sybrenstuvel** [OpenStitching/stitching#79] [Q&A]
Hello!

Is it possible to use `stitching` to generate 360x180 degree equirectangular images?

My context: I like shooting [panorama photos](https://stuvelfoto.nl/panorama/), and I'm looking for alternatives to Hugin to merge them together. Right now I use a two-stage approach, where I first merge the 7 bracketed exposures of each camera position, and then stitch the resulting 38 photos together. The exposure merging I managed to write some software for to automate ([brakketor](https://gitlab.com/dr.sybren/brakketor)), and I'm searching for something to help me automate the stitching as well.

Cheers,
Sybren

**Comment by lukasalexanderweber** [comment]
Hey,

your work looks pretty awesome!

Probably it would bei best if you share a use case with resulting Images from brakkeror.

Cheers 
Lukas

**Reply by bakermanbrian** [reply]
@lukasalexanderweber so this stitching library won't be a great use case if I'm looking for equirectangular stitching is that right?

**Comment by daduke** [comment]
I'll have to play around with this as well, right now I'm also using my [own bunch of scripts](https://daduke.org/linux/hdrpano.html) to generate my panoramas. Nice work!

**Comment by sybrenstuvel** [comment]
Thanks! I'll downscale some images & attach an example (the 38 TIFFs I get from Brakketor are now around 7 GB in total).

**Comment by sybrenstuvel** [comment]
It took a while (sorry about that), but here's a set of JPEG files that form a single SDR panorama: https://stack.stuvel.eu/s/u9gJD01ZrEAb7iT2 (316 MB). I think these are pretty representative of what I typically shoot, albeit a little simple in the exposure (there aren't any big blown-out or full-shadow areas).

Is it currently possible to let Sticher write out the transforms for each image, and use those in a new run to place another set of images at exactly the same spot? That would help tremendously when stitching HDR images, as not all HDR layers will have enough detail in all the photos to build the same correspondences.

**Reply by lukasalexanderweber** [reply]
> It took a while (sorry about that), but here's a set of JPEG files that form a single SDR panorama: https://stack.stuvel.eu/s/u9gJD01ZrEAb7iT2 (316 MB). I think these are pretty representative of what I typically shoot, albeit a little simple in the exposure (there aren't any big blown-out or full-shadow areas).

Thanks alot for sharing the images! At the moment, I want to restructure the discussion section to provide a FAQ and also make sure most use cases can be reproduces, ideally by collecting the images in a seperate repository. You might want to consider uploading your images into https://github.com/OpenStitching/community-images so I can make a long lasting link between the discussion and the images.

> Is it currently possible to let Sticher write out the transforms for each image, and use those in a new run to place another set of images at exactly the same spot? That would help tremendously when stitching HDR images, as not all HDR layers will have enough detail in all the photos to build the same correspondences.

What I have previously done was to [pickle](https://docs.python.org/3/library/pickle.html) the cameras to apply them onto another image set. What people use in [video stitching](https://stackoverflow.com/questions/77639112/stitching-of-video-frames-from-two-calibrated-cctv-cameras) is register the cameras once and apply those onto all frames that follow.

**Comment by sybrenstuvel** [comment]
> Thanks alot for sharing the images! At the moment, I want to restructure the discussion section to provide a FAQ and also make sure most use cases can be reproduces, ideally by collecting the images in a seperate repository. You might want to consider uploading your images into https://github.com/OpenStitching/community-images so I can make a long lasting link between the discussion and the images.

Done: https://github.com/OpenStitching/images/pull/1

> What I have previously done was to [pickle](https://docs.python.org/3/library/pickle.html) the cameras to apply them onto another image set. What people use in [video stitching](https://stackoverflow.com/questions/77639112/stitching-of-video-frames-from-two-calibrated-cctv-cameras) is register the cameras once and apply those onto all frames that follow.

Thanks!

I want to respectfully point out that my original question still hasn't been answered:

> Is it possible to use `stitching` to generate 360x180 degree equirectangular images?

Since spherical panoramas have more requirements, like the left and right side of the resulting image connecting to each other, I don't want to just assume it's going to work. However, since nobody told me yet that it's not supported, I'm hopeful.

**Reply by lukasalexanderweber** [reply]
> I want to respectfully point out that my original question still hasn't been answered:

there is another discussion https://github.com/OpenStitching/stitching/discussions/89 in which the requirement `left and right side of the resulting image connecting to each other` is shown. Unfortunately we havn't come up with a sollution yet. I havn't spend much time with equirectangular stitching and don't know much about OpenCVs warpers. If OpenCV don't work out of the box, maybe we could include some logic or pre/postprocessing within `stitching`.

As I pointed out in https://github.com/OpenStitching/stitching/discussions/89:

> There is definetly need for this:
> 
> https://stackoverflow.com/questions/69043002/360-equirectangular-image-stitching
> 
> https://www.reddit.com/r/opencv/comments/ph1xw8/question_equirectangular_image_stitching_method/
> 
> https://answers.opencv.org/question/27176/can-opencv-stitcher-generate-a-full-spherical-panoramas-like-street-view/
> 
> It would be great to get this working and provide an example!
